src/Context/AuthContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { jwtDecode } from "jwt-decode";
+import { createContext, useEffect, useState } from "react";
+
+export const AuthContext = createContext(null);
+
+
+export default function AuthContextProvider(props) {
+
+  const [loginData, setLoginData] = useState(null)
+
+  const saveLoginData = () => {
+
+    const encodedToken = localStorage.getItem('token')
+
+    const decodedToken = jwtDecode(encodedToken)
```

Comment:
We can use the token and id coming from the login and call the api that return the user details instead of using jwtDecode, as the decoded data will not contain the image url which we will need to display in the ui.

Reply:
Ok , we will use jwtcode only in navbar just for take the user id and send it through API to get user information

Reply:
My intention was that there is no need for an external package like jwtDecode, as the data is already provided by the backend. Using external package to deal with sensitive data like the token might be a security risk.